lib/puppet/type/packageprovider.rb

```diff
@@ -0,0 +1,30 @@
+Puppet::Type.newtype(:packageprovider) do
+  @doc = 'Manage PowerShell Package providers for PowerShell modules'
+
+  newproperty(:ensure) do
+    newvalue(:present) do
+      provider.create
```

Comment:
We cannot delete packageproviders or modify them once created, as there is no external official API to do so